File: src/cpp/tests/spatial_metrics_public_test.cpp
#ifdef GTEST_AVAILABLE
#include <gtest/gtest.h>
#endif

#include <vector>
#include "Region.h"

#ifdef GTEST_AVAILABLE
TEST(SpatialMetricsPublic, FromImage) {
    grownet::Region region("spatial-metrics");
    std::vector<std::vector<double>> img{{0.0, 1.0, 0.0}, {0.0, 0.0, 0.0}};
    auto m = region.computeSpatialMetrics(img, /*preferOutput*/ false);
    EXPECT_EQ(m.getActivePixels(), 1);
    EXPECT_DOUBLE_EQ(m.getCentroidRow(), 0.0);
    EXPECT_DOUBLE_EQ(m.getCentroidCol(), 1.0);
    EXPECT_EQ(m.getBboxRowMin(), 0);
    EXPECT_EQ(m.getBboxRowMax(), 0);
    EXPECT_EQ(m.getBboxColMin(), 1);
    EXPECT_EQ(m.getBboxColMax(), 1);
}
#endif

File: src/java/tests/ai/nektron/grownet/tests/SpatialMetricsPublicTest.java
package ai.nektron.grownet.tests;

import ai.nektron.grownet.Region;
import ai.nektron.grownet.metrics.RegionMetrics;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class SpatialMetricsPublicTest {

    @Test
    public void computeSpatialMetricsFromImage() {
        Region region = new Region("spatial-metrics");
        double[][] img = new double[][]{
                {0.0, 1.0, 0.0},
                {0.0, 0.0, 0.0}
        };
        RegionMetrics m = region.computeSpatialMetrics(img, /*preferOutput*/ false);
        assertEquals(1L, m.getActivePixels());
        assertEquals(0.0, m.getCentroidRow(), 1e-12);
        assertEquals(1.0, m.getCentroidCol(), 1e-12);
        assertEquals(0, m.getBboxRowMin());
        assertEquals(0, m.getBboxRowMax());
        assertEquals(1, m.getBboxColMin());
        assertEquals(1, m.getBboxColMax());
    }
}

